src/main/java/com/commercetools/config/PaypalPlusStartupConfiguration.java

```diff
@@ -0,0 +1,97 @@
+package com.commercetools.config;
+
+import com.commercetools.pspadapter.facade.PaypalPlusFacade;
+import com.commercetools.pspadapter.facade.PaypalPlusFacadeFactory;
+import com.commercetools.pspadapter.notification.validation.NotificationValidationFilter;
+import com.commercetools.pspadapter.notification.validation.NotificationValidationInterceptor;
+import com.commercetools.pspadapter.tenant.TenantConfig;
+import com.commercetools.pspadapter.tenant.TenantConfigFactory;
+import com.commercetools.pspadapter.tenant.TenantProperties;
+import com.paypal.api.payments.Webhook;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurerAdapter;
+
+import javax.annotation.Nonnull;
+import javax.servlet.Filter;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static com.commercetools.payment.constants.Psp.NOTIFICATION_PATH_URL;
+import static java.lang.String.format;
+
+@Configuration
+public class PaypalPlusStartupConfiguration extends WebMvcConfigurerAdapter {
```

Comment:
But Spring resolves these dependencies, right?

Reply:
right, but if you read thoroughly:

> This approach simplifies container instantiation, as only one class needs to be dealt with, rather than requiring the developer to remember a potentially large number of `@Configuration` classes during construction.

Reply:
still missing

Reply:
https://github.com/commercetools/commercetools-paypal-plus-integration/pull/57/commits/f33f0cb3b6323e6607827a73aea7f9befee1b63b#diff-3ef6a71ca2b858b619b09d307bf86e16R18

Reply:
ha, it looks like I did it the other way around 😄  or you did it the other way around.

Reply:
actually, should be opposite:
62d30eb...bae9054

Reply:
> ha, it looks like I did it the other way around 😄 or you did it the other way around.

apparently `PaypalPlusStartupConfiguration` is dependent on `ApplicationConfiguration`, but not opposite. So, there is only one proper way to do this :)